Fixes Intellij builds jars for java 9/11/13

teyrana opened the pull request:
no source code changes, just adding the right libraries to the right Intellij *.iml, *.iml project files.
This also  allows intellij to generate a release jar.

teyrana commented:
@wolsen 
you've used Intellij before... do you have a minute to see if this works on your machine, real quick?

wolsen commented:
@teyrana I haven’t used IntelliJ before, but I’ll give it a go tomorrow

teyrana commented:
oh!  My bad. Which build system do you use, then?

teyrana commented:
@neilbalch 
Hey, I know you're busy, but you used intellij, right? Would you have a few minutes to pull this pr, and see if it builds for you in Intellij?

wolsen commented:
> oh! My bad. Which build system do you use, then?

I ant, vim and a touch of eclipse - just downloaded intellij, will let you know

wcravens commented:
I tried with a completely fresh installation of IntelliJ Community 2019.3.3 on Ubuntu 18.04.  I did literally nothing besides pull the branch and then Build->Build Project in IntelliJ and got the included error.  I'm sure there is some configuration that I need.  Are there instructions somewhere?

![Screenshot from 2020-03-14 12-51-11](https://user-images.githubusercontent.com/1262847/76687521-cd5aea80-65f2-11ea-8a01-71c243721a83.png)

wolsen commented:
I ran into the same issue as @wcravens as well and had to specify a valid SDK in the Project settings. Once I did that, I have compilation issues due to not referencing the various jars in the lib-test project. I'm not deeply familiar with IntelliJ, so it could be something I'm doing wrong on my end.

teyrana commented:
Unfortunately, you have to manually set that part.   

Just point intellij at your currently installed jdk (like openjdk11 ;)

1. Open the 'File' Menu > 'Project Structure ...' 
2. in the dialog that pops up, look at the 'Platform Settings > SDKs' option in the left menu.
3. in the second column (which may be empty?) click the '+' to add an sdk.  Point it at your installed sdk location
4. if there's a red / broken sdk in there currently, the project expects an SDK named 'OpenJDK 11'.  Merely fixing the path-broken one may let you build....
5. otherwise, select 'Project Settings > Project' on the left menu, and set the 'Project SDK' option.

teyrana commented:
huh. okay, now I'm having the same issue. :P

teyrana commented:
okay, pushed a new version.   
The goal is that it should work out-of-the-box, once you give it a jdk.

wolsen commented:
I still needed to make a couple of changes with the new revision:

1. Add all the jars in the core/lib dir as extracted resources to the artifact definition
2. Add a library of the core/lib dir as a dependency for the submodules

The output of git diff for my local changes is located [here](https://paste.ubuntu.com/p/HmQ3cysTfD/) (as captured by git diff | pastebinit -f diff) 

I also note that the resultant OpenRocket.jar artifact is different than the one produced by the ant build. The artifact explodes the relevant jar files and includes them. I'm not entirely sure of all the drawbacks with this approach, but if all is well then perhaps the ant build should use this method as well.

wcravens commented:
Mine errors out because of a missing JUnit.  'java: package org.junit does not exist'

wcravens commented:
I also just had to switch to a different Java 11 in the Project.  IntelliJ offered me 11.0.6 and language level of 7.

I'm not trying to work around any of these issues myself because I figure it's a better test to not touch anything unless the team gives instructions.  That was we are testing it as vanilla as possible.

neilbalch commented:
Sure thing! Still have to configure the JDK for the project, and intellij has some build errors for you, @teyrana :

```
Information:java: Errors occurred while compiling module 'OpenRocket Core'
Information:javac 11 was used to compile java sources
Information:Module "OpenRocket Core" was fully rebuilt due to project configuration/dependencies changes
Information:3/14/2020 13:23 - Build completed with 100 errors and 30 warnings in 25 s 525 ms
C:\Users\neilb\Documents\GitHub\openrocket\core\src\de\congrace\exp4j\CommandlineInterpreter.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\de\congrace\exp4j\ExpressionBuilder.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\rocketcomponent\ReferenceType.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\rocketcomponent\FinSet.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\aerodynamics\BarrowmanCalculator.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\simulation\SimulationStatus.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\DatabaseMotorFinder.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\openrocket\importt\ComponentPresetSetter.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\rocketcomponent\BodyTube.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\InnerBodyTubeDTO.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\optimization\rocketoptimization\domains\StabilityDomain.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\optimization\rocketoptimization\parameters\StabilityParameter.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\preset\loader\RocksimComponentFileLoader.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\rocketcomponent\RocketUtils.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\simulation\BasicEventSimulationEngine.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\simulation\listeners\example\RollControlListener.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\unit\CaliberUnit.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\optimization\rocketoptimization\modifiers\FlightConfigurationModifier.java
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\aerodynamics\BarrowmanCalculatorTest.java
Error:(3, 24) java: package org.junit does not exist
Error:(3, 1) java: static import only from classes and interfaces
Error:(4, 24) java: package org.junit does not exist
Error:(4, 1) java: static import only from classes and interfaces
Error:(5, 24) java: package org.junit does not exist
Error:(5, 1) java: static import only from classes and interfaces
Error:(7, 17) java: package org.junit does not exist
Error:(8, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\aerodynamics\FinSetCalcTest.java
Error:(3, 24) java: package org.junit does not exist
Error:(3, 1) java: static import only from classes and interfaces
Error:(4, 17) java: package org.junit does not exist
Error:(5, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\aerodynamics\SymmetricComponentCalcTest.java
Error:(3, 24) java: package org.junit does not exist
Error:(3, 1) java: static import only from classes and interfaces
Error:(4, 17) java: package org.junit does not exist
Error:(5, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\arch\TestSystemInfo.java
Error:(3, 24) java: package org.junit does not exist
Error:(3, 1) java: static import only from classes and interfaces
Error:(7, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\communication\BugReportTest.java
Error:(3, 24) java: package org.junit does not exist
Error:(3, 1) java: static import only from classes and interfaces
Error:(4, 24) java: package org.junit does not exist
Error:(4, 1) java: static import only from classes and interfaces
Error:(5, 24) java: package org.junit does not exist
Error:(5, 1) java: static import only from classes and interfaces
Error:(6, 24) java: package org.junit does not exist
Error:(6, 1) java: static import only from classes and interfaces
Error:(12, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\communication\UpdateInfoTest.java
Error:(3, 24) java: package org.junit does not exist
Error:(3, 1) java: static import only from classes and interfaces
Error:(4, 24) java: package org.junit does not exist
Error:(4, 1) java: static import only from classes and interfaces
Error:(5, 24) java: package org.junit does not exist
Error:(5, 1) java: static import only from classes and interfaces
Error:(6, 24) java: package org.junit does not exist
Error:(6, 1) java: static import only from classes and interfaces
Error:(7, 24) java: package org.junit does not exist
Error:(7, 1) java: static import only from classes and interfaces
Error:(8, 24) java: package org.junit does not exist
Error:(8, 1) java: static import only from classes and interfaces
Error:(19, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\util\BaseTestCase\BaseTestCase.java
Error:(9, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\database\ThrustCurveMotorSetTest.java
Error:(3, 24) java: package org.junit does not exist
Error:(3, 1) java: static import only from classes and interfaces
Error:(4, 24) java: package org.junit does not exist
Error:(4, 1) java: static import only from classes and interfaces
Error:(5, 24) java: package org.junit does not exist
Error:(5, 1) java: static import only from classes and interfaces
Error:(6, 24) java: package org.junit does not exist
Error:(6, 1) java: static import only from classes and interfaces
Error:(7, 24) java: package org.junit does not exist
Error:(7, 1) java: static import only from classes and interfaces
Error:(18, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\file\iterator\TestDirectoryIterator.java
Error:(7, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\file\iterator\TestFileIterator.java
Error:(3, 24) java: package org.junit does not exist
Error:(10, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\file\motor\TestMotorLoader.java
Error:(3, 24) java: package org.junit does not exist
Error:(3, 1) java: static import only from classes and interfaces
Error:(4, 24) java: package org.junit does not exist
Error:(4, 1) java: static import only from classes and interfaces
Error:(5, 24) java: package org.junit does not exist
Error:(5, 1) java: static import only from classes and interfaces
Error:(15, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\file\openrocket\importt\DocumentConfigTest.java
Error:(3, 24) java: package org.junit does not exist
Error:(3, 1) java: static import only from classes and interfaces
Error:(7, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\file\openrocket\OpenRocketSaverTest.java
Error:(3, 24) java: package org.junit does not exist
Error:(3, 1) java: static import only from classes and interfaces
Error:(4, 24) java: package org.junit does not exist
Error:(4, 1) java: static import only from classes and interfaces
Error:(5, 24) java: package org.junit does not exist
Error:(5, 1) java: static import only from classes and interfaces
Error:(38, 17) java: package org.junit does not exist
Error:(39, 17) java: package org.junit does not exist
Error:(40, 17) java: package org.junit does not exist
Error:(265, 24) java: cannot access javax.inject.Provider
  class file for javax.inject.Provider not found
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\file\rocksim\export\RocksimDocumentDTOTest.java
Error:(15, 17) java: package org.junit does not exist
Error:(16, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\file\rocksim\importt\RocksimTestBase.java
Error:(12, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\file\rocksim\importt\BodyTubeHandlerTest.java
Error:(13, 17) java: package org.junit does not exist
Error:(14, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\file\rocksim\importt\FinSetHandlerTest.java
Error:(6, 24) java: package org.junit does not exist
Error:(6, 1) java: static import only from classes and interfaces
Error:(7, 24) java: package org.junit does not exist
Error:(7, 1) java: static import only from classes and interfaces
Error:(8, 24) java: package org.junit does not exist
Error:(8, 1) java: static import only from classes and interfaces
Error:(14, 17) java: package org.junit does not exist
Error:(15, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\file\rocksim\importt\InnerBodyTubeHandlerTest.java
Error:(13, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\file\rocksim\importt\LaunchLugHandlerTest.java
Error:(14, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\file\rocksim\importt\MassObjectHandlerTest.java
Error:(13, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\file\rocksim\importt\NoseConeHandlerTest.java
Error:(15, 17) java: package org.junit does not exist
Error:(16, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\file\rocksim\importt\ParachuteHandlerTest.java
Error:(8, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\file\rocksim\importt\RingHandlerTest.java
Error:(8, 17) java: package org.junit does not exist
Error:(9, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\file\rocksim\importt\RocksimLoaderTest.java
Error:(11, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\file\rocksim\importt\StreamerHandlerTest.java
Error:(8, 17) java: package org.junit does not exist
Error:(9, 17) java: package org.junit does not exist
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\rocketcomponent\LaunchLugTest.java
C:\Users\neilb\Documents\GitHub\openrocket\core\test\net\sf\openrocket\rocketcomponent\FlightConfigurationTest.java
```

However, it may be of note that `openrocket/openrocket:unstable` fails to build with the same errors.

neilbalch commented:
It should be noted that my old `neilbalch/openrocket:java11-upgrade` branch still builds fine.

teyrana commented:
> Add all the jars in the core/lib dir as extracted resources to the artifact definition

got it. added those to the jar artifact file.

> Add a library of the core/lib dir as a dependency for the submodules

okay, i've added the javax library back in.  I think it got lost in all the rebases. :P


> I also note that the resultant OpenRocket.jar artifact is different than the one produced by the ant build. 

Yea, I see that.   I'm honestly not sure if it's better or worse?  And until someone finds an issue, I'll just leave it alone.  

Besides, we can always build the 'official' jars from the known "best" method if people want.

neilbalch commented:
Not sure if you're still looking for my input here, but this is what results from building your latest three commits:

```
Information:java: Errors occurred while compiling module 'OpenRocket Core'
Information:javac 13.0.1 was used to compile java sources
Information:3/14/2020 14:26 - Build completed with 100 errors and 31 warnings in 7 s 771 ms
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\AbstractTransitionDTO.java
Error:(18, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(19, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(20, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(21, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(22, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(23, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\BasePartDTO.java
Error:(3, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(4, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(5, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(6, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\BodyTubeDTO.java
Error:(20, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(21, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(22, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(23, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(24, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(25, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(26, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\BulkheadDTO.java
Error:(6, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(7, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(8, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\CenteringRingDTO.java
Error:(7, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(8, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(9, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(10, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(11, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\CustomFinSetDTO.java
Error:(7, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(8, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(9, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(10, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\FinSetDTO.java
Error:(10, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(11, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(12, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(13, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\EngineBlockDTO.java
Error:(5, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(6, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(7, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\InnerBodyTubeDTO.java
Error:(5, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(6, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(7, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Warning:(112, 41) java: getLocations() in net.sf.openrocket.rocketcomponent.RocketComponent has been deprecated
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\LaunchLugDTO.java
Error:(3, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(4, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(5, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(6, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\MassObjectDTO.java
Error:(6, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(7, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(8, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(9, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\NoseConeDTO.java
Error:(6, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(7, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(8, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(9, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\ParachuteDTO.java
Error:(7, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(8, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(9, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(10, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\RocketDesignDTO.java
Error:(3, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(4, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(5, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\StageDTO.java
Error:(11, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(12, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(13, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(14, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\TransitionDTO.java
Error:(6, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(7, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(8, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(9, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\RocksimDesignDTO.java
Error:(5, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(6, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(7, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\RocksimDocumentDTO.java
Error:(5, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(6, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(7, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(8, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\RocksimSaver.java
Error:(9, 17) java: package javax.xml.bind is not visible
  (package javax.xml.bind is declared in module java.xml.bind, which is not in the module graph)
Error:(10, 17) java: package javax.xml.bind is not visible
  (package javax.xml.bind is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\StreamerDTO.java
Error:(6, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(7, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(8, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(9, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\TubeCouplerDTO.java
Error:(6, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(7, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(8, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\rocksim\export\TubeFinSetDTO.java
Error:(6, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(7, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(8, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(9, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\preset\xml\BaseComponentDTO.java
Error:(10, 17) java: package javax.xml.bind is not visible
  (package javax.xml.bind is declared in module java.xml.bind, which is not in the module graph)
Error:(11, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(12, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(13, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(14, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(15, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(16, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(17, 33) java: package javax.xml.bind.annotation.adapters is not visible
  (package javax.xml.bind.annotation.adapters is declared in module java.xml.bind, which is not in the module graph)
Error:(18, 33) java: package javax.xml.bind.annotation.adapters is not visible
  (package javax.xml.bind.annotation.adapters is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\preset\xml\MaterialDTO.java
Error:(3, 17) java: package javax.xml.bind is not visible
  (package javax.xml.bind is declared in module java.xml.bind, which is not in the module graph)
Error:(4, 17) java: package javax.xml.bind is not visible
  (package javax.xml.bind is declared in module java.xml.bind, which is not in the module graph)
Error:(5, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(6, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
Error:(7, 22) java: package javax.xml.bind.annotation is not visible
  (package javax.xml.bind.annotation is declared in module java.xml.bind, which is not in the module graph)
C:\Users\neilb\Documents\GitHub\openrocket\core\src\de\congrace\exp4j\Variable.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\de\congrace\exp4j\CommandlineInterpreter.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\de\congrace\exp4j\ExpressionBuilder.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\rocketcomponent\ReferenceType.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\rocketcomponent\FinSet.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\aerodynamics\BarrowmanCalculator.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\document\Simulation.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\simulation\customexpression\CustomExpression.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\simulation\SimulationStatus.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\DatabaseMotorFinder.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\openrocket\importt\ComponentPresetSetter.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\rocketcomponent\BodyTube.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\file\simplesax\SimpleSAX.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\optimization\rocketoptimization\domains\StabilityDomain.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\optimization\rocketoptimization\parameters\StabilityParameter.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\preset\loader\RocksimComponentFileLoader.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\rocketcomponent\RocketUtils.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\simulation\BasicEventSimulationEngine.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\simulation\listeners\example\RollControlListener.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\unit\CaliberUnit.java
C:\Users\neilb\Documents\GitHub\openrocket\core\src\net\sf\openrocket\optimization\rocketoptimization\modifiers\FlightConfigurationModifier.java
```

teyrana commented:
@neilbalch 
OH.  Huh I reproduced this by performing a fresh checkout. 
I'll let you know when I've added a fix.

teyrana commented:
@neilbalch 
Okay, I reworked how libraries were included.   If you pull the branch again, 
[1309dd2982](https://github.com/teyrana/openrocket/commit/1309dd2982f40c71fa18ca9e1e18336b48b46a57) should fix the issue.

neilbalch commented:
@teyrana Success! Other than a couple of warnings for which I can submit a patch: 
![image](https://user-images.githubusercontent.com/3847882/76804466-8fe49180-6799-11ea-9b0a-7110b258383e.png)

Looks great!

![image](https://user-images.githubusercontent.com/3847882/76804500-aa1e6f80-6799-11ea-96e5-ee3762027c7e.png)

neilbalch commented:
@teyrana PR sent
